l10n_fi_reconcile/tests/test_reconciliation_proposition.py

```diff
@@ -0,0 +1,141 @@
+            'payment_reference': REFERENCE_ONE_WITH_ZEROS,
+        })
+        invoices[REFERENCE_TWO_WITH_ZEROS] = self.env['account.invoice'].create({
+            'partner_id': self.partner.id,
+            'account_id': self.account.id,
+            'journal_id': self.journal.id,
+            'company_id': self.company.id,
+            'payment_reference': REFERENCE_TWO_WITH_ZEROS,
+        })
+
+        self.account_bank_statement_id = self.env['account.bank.statement'].create({
+            'journal_id': self.journal.id,
+            'company_id': self.company.id,
+        })
+
+        # We need a payment_method to assign to payments. This is how account.invoice
+        # gets payment_method when creating payments
+        payment_method = self.env.ref('account.account_payment_method_manual_in')
+        self.assertTrue(payment_method)
+        # We need to create move lines at the same time so the journal stays in balance,
+        # so create a list of tuples with dicts that hold the values and pass them to the account.move
+        # line_ids which will then link and create them all at once.
+        # [(0, 0, {values}), (0, 0, ...]
+        aml_lines = []
+        for line in ACCOUNT_MOVE_LINES:
+            # we just need a payment_id to give to account move lines. Contains no relevant information,
+            # just enough values to be able to create the payment
+            payment_id = self.env['account.payment'].create({
+                'amount': line['amount'],
+                'partner_id': self.partner.id,
+                'partner_type': 'customer',
+                'journal_id': self.journal.id,
+                'payment_type': 'inbound',
+                'payment_method_id': payment_method.id,
+            })
+
+            # Debit and Credit lines to keep the move in balance.
+            aml_lines.append((0, 0, {
+                'name': line['name'],
+                'account_id': self.account.id,
+                'date_maturity': fields.Date.today(),
+                # account.move.line gets its payment_reference from the linked invoice.
+                'invoice_id': invoices[line['payment_reference']].id,
+                'debit': line['amount'],
+                'payment_id': payment_id.id,
+            }))
+            aml_lines.append((0, 0, {
+                'name': line['name'],
+                'account_id': self.account.id,
+                'date_maturity': fields.Date.today(),
+                'invoice_id': invoices[line['payment_reference']].id,
+                'credit': line['amount'],
+                'payment_id': payment_id.id,
+            }))
+
+        self.account_move_id = self.env['account.move'].create({
+            'name': 'testMove1',
+            'line_ids': aml_lines,
+            'company_id': self.company.id,
+            'journal_id': self.journal.id,
+        })
+
+    def test_cases(self):
+        for i, case in enumerate(BANK_STATEMENT_LINES):
```

Comment:
Do these tests also cover the case of Odoo trying to automatically reconcile statements? In those cases (as I understand it), Odoo should only find perfect matches, and disregard imperfect ones.